storage/bootstrap/bootstrapper/commitlog/source.go

```diff
@@ -49,9 +54,50 @@ type encoder struct {
 	enc         encoding.Encoder
 }
 
-type encoderMap struct {
-	id       ts.ID
-	encoders map[time.Time][]encoder
+type encodersAndRanges struct {
```

Comment:
super nit: could you move the helper type decls and methods to the bottom of the file. Couple of rules of thumb (apply with discretion obv) -
- Try to have related code close by in a single file (i.e. two methods on the same object, or the exported+helper methods)
- Order the contents of a file's types/methods in decreasing order of importance (so primary type/method at the top, helper types/methods at the bottom).

Reply:
fixed